watchtower/__init__.py

```diff
@@ -249,11 +254,15 @@ def __init__(
         # Creating the client should be the final call in __init__, after all instance attributes are set.
         # This ensures that failing to create the session will not result in any missing attribtues.
         if boto3_client is None and boto3_profile_name is None:
```

Comment:
Can you add some unit tests that assert these 4 branches (that the client has the custom end point when expected and throws a warning when expected, etc).